src/playlist-controller.js

```diff
@@ -2024,6 +2025,21 @@ export class PlaylistController extends videojs.EventTarget {
     return Config.BUFFER_HIGH_WATER_LINE;
   }
 
+  addDaterangeToTextTrack(metadataArray) {
+    const timestampOffset = this.sourceUpdater_.videoBuffer ?
+      this.sourceUpdater_.videoTimestampOffset() : this.sourceUpdater_.audioTimestampOffset();
```

Comment:
Is it possible to have `dateRanges` from multiple discontinuities?

Reply:
More context: I see we are using `timestampOffset` discontinuities will have different `timestampOffset`s.

My thoughts: I am not sure we should use timestampOffset for dateRanges